### PR TITLE
[DEV-3178] BambooHR: added terms and conditions slot to register

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> Added ability to include slotted content on \<sqm-portal-register> using `slot="terms"`
  - Slotted content is displayed above the registration button

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-3178
 - Process.st launch checklist: https://app.process.st/runs/DEV-3178-BambooHR-Registration-Update-ijmqc50i69T6Rn53_69AxQ

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist
- [x] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.feature [hunk 1]
```diff
@@ -99,4 +99,21 @@ Feature: Portal Register
         Examples:
             | mayHave      | value          | redirectPath   |
             | doesn't have | N/A            | /verifyEmail   |
-            | has          | /verifyMyEmail | /verifyMyEmail |
\ No newline at end of file
+            | has          | /verifyMyEmail | /verifyMyEmail |
+
+    @ui
+    Scenario Outline: Terms and conditions link can be included above the register button
```
Comment (DerekSiemens) [inline packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.feature:105]:
```suggestion
    Scenario Outline: Slotted content can be included above the register button
```
Scenario description should be more generic, while we are building it for T&C text/link, the slot we are adding is  much more general than that

File: packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.feature [hunk 2]
```diff
@@ -99,4 +99,21 @@ Feature: Portal Register
...
+            <p slot="terms">
+            By signing up you agree to the
+            <a href="https://example.com" target="_blank">
+            Terms and Conditions
+            </a>
+            </p>
+            </sqm-portal-register>
+            """
+        Then the terms and conditions are shown above the register button
+        And the link will open in a new tab
```
Comment (DerekSiemens) [inline packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.feature:119]:
```suggestion
        And the link opens in a new tab
```

File: packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.feature [hunk 3]
```diff
@@ -99,4 +99,21 @@ Feature: Portal Register
...
+            <sqm-portal-register>
+            <p slot="terms">
+            By signing up you agree to the
+            <a href="https://example.com" target="_blank">
+            Terms and Conditions
+            </a>
+            </p>
+            </sqm-portal-register>
+            """
+        Then the terms and conditions are shown above the register button
```
Comment (DerekSiemens) [inline packages/mint-components/src/components/sqm-portal-register/sqm-portal-register.feature:118]:
```suggestion
        Then the terms and conditions slotted content is shown above the register button
```

**Comment by DerekSiemens** [comment]
Missing changelog update!

File: packages/mint-components/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@saasquatch/mint-components",
-  "version": "1.4.0",
```
Comment (DerekSiemens) [inline packages/mint-components/package.json:3]: Flagging that 1.4.1 is also being used by https://github.com/saasquatch/program-tools/pull/246

Comment (00salmon) [reply packages/mint-components/package.json:3]: ah was wondering what was deployed to that version, thank you!

**Comment by 00salmon** [comment]
> Missing changelog update!

Changelog updated, but due to versioning we will need to officially merge + deploy this once the Clara PR is pushed and published.

**Review by loganvolkers — DISMISSED** [review]
Looks good to me

**Review by zachharrison — APPROVED** [review]

**Pull request merged**